extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/runtime/OidcClientImpl.java

```diff
@@ -84,7 +87,13 @@ public OidcClientImpl(WebClient client, String tokenRequestUri, String grantType
                     body.add(OidcConstants.CLIENT_ASSERTION_TYPE, OidcConstants.JWT_BEARER_CLIENT_ASSERTION_TYPE);
                     body.add(OidcConstants.CLIENT_ASSERTION, OidcCommonUtils.signJwtWithKey(oidcConfig, clientJwtKey));
                 }
-                return request.sendForm(body).onItem()
+                // Retry up to 3 seconds if the connection is closed
+                Uni<HttpResponse<Buffer>> response = request.sendForm(body)
+                        .onFailure(ConnectException.class)
+                        .retry()
+                        .withBackOff(REQUEST_RETRY_BACKOFF_DURATION, REQUEST_RETRY_BACKOFF_DURATION)
+                        .expireIn(3);
```

Comment:
If you say 'expireIn' you may discard the back off configuration. Maybe atMost is better. 
Or make expireIn configurable.

Reply:
Hi @cescoffier, thanks, I was thinking to retry for up to 3 times with a 1 sec delay - I'm hoping it will be a fairly rare situation where the connection will drop :-), so for now may be I can avoid adding one more configuration property (we already have 1 property to configure an initial connection delay) - to avoid users putting a large enough expiry  value which can cause the client to wait for a while if something goes wrong with the connection - but if it is a transient problem then 3 secs should be enough I guess.
So if I just do `retry().expireIn(3);` will it pause for `1sec`  between retries by default ?
I'll check `atMost` as well

Reply:
Yeah, it is a 1 sec backoff by default, `atMost` looks good

Reply:
I believe atMost is what you want here.